orca-queue/src/main/kotlin/com/netflix/spinnaker/orca/q/handler/ExpressionAware.kt

```diff
@@ -60,12 +62,28 @@ interface ExpressionAware {
         return result
       }
     }
+
+    // Clean up errors: since expressions are evaluated multiple times, it's possible that when
+    // they were evaluated before the execution started not all data was available and the evaluation failed for
+    // some property. If that evaluation subsequently succeeds, make sure to remove past error messages from the
+    // context. Otherwise, it's very confusing in the UI because the value is clearly correctly evaluated but
+    // the error is still shown
+    if (hasFailedExpressions()) {
+      val failedExpressions = this.context[PipelineExpressionEvaluator.SUMMARY] as MutableMap<String, *>
+
+      failedExpressions.keys.forEach( { expressionKey ->
+        if (evalSummary.wasAttempted(expressionKey) && !evalSummary.hasFailed(expressionKey)) {
+          failedExpressions.remove(expressionKey)
+        }
+      })
```

Comment:
Just a small little thing here, final arguments in Kotlin into a method don't need to be wrapped in parens.

Reply:
thanks!